src/main/java/io/anserini/util/TrainingDataGenerator.java

```
@@ -0,0 +1,360 @@
+   */
+  static String extractValueFromTypedLiteralString(String literalString) {
+    return NTriplesUtil.parseLiteral(literalString, valueFactory).stringValue();
+  }
+
+  /**
+   * Helper function. Converts freebase URI to freebase mention id
+   *
+   * @param freebaseUri freebase uri, similar to
+   * @return freebase mention id
+   */
+  static String freebaseUriToFreebaseId(String freebaseUri) {
+    return freebaseUri.substring(freebaseUri.lastIndexOf('/')).replace('.', '/');
+  }
+
+  /**
+   * Writes an array as a line separated by TRAINING_DATA_SEPARATOR.
+   *
+   * @param l    the logger to write to (the training data file)
+   * @param data array of data
+   */
+  void writeToTrainingFiles(Logger l, String... data) {
+    StringBuilder sb = new StringBuilder();
+    for (int i = 0; i < data.length; i++) {
+      if (i > 0)
+        sb.append(TRAINING_DATA_SEPARATOR);
+
+      if (data[i].contains(TRAINING_DATA_SEPARATOR))
+        sb.append("\"").append(data[i]).append("\"");
+      else
+        sb.append(data[i]);
+    }
+
+    l.info(sb.toString());
+  }
+
+  /**
+   * Determine which properties to retrieve and runs the training data retrieval pipeline.
+   *
+   * @throws Exception on error
+   */
+  private void generateTrainingData() throws Exception {
+    switch (args.property.toLowerCase()) {
+      case "birthdate":
+        birthdate();
+        break;
+
+      default:
+        LOG.error("Cannot generate training data for property: {}", args.property);
+        throw new IllegalArgumentException("Cannot generate training data for property: " + args.property + ". Supported properties are: " + supportedPropertiesStr);
+    }
+  }
+
+  /**
+   * Generate training data for property birth date
+   * <p>
+   * Note: this function might need some refactoring when we add more properties
+   */
+  void birthdate() throws Exception {
+    QueryParser queryParser = new QueryParser(FIELD_BIRTHDATE, getKbIndexAnalyzer());
```

can we throw a specific exception